### ISSUE TITLE
Support tcp instead Unix socket

### ISSUE THREAD

**Issue by mvgijssel** [superbrothers/opener#6]
Thanks for making this awesome tool! I was trying to get this work by having a Docker container open a webpage on the host machine. Because I’m on macOS docker is unable to share a Unix socket (https://github.com/docker/for-mac/issues/483#issuecomment-620229892), but can communicate through regular TCP. 

Therefore I’m wondering if it’s possible to add support for running the opener daemon on a TCP port?

**Comment by superbrothers** [comment]
Thanks for trying my tool.

I can provide support for TCP, but first I want to clarify what you are trying to achieve. Do you want to use `open` command in a container to open a URL in the host machine's browser?

```
┌────────────────────┐
│                    │
│ ┌────────────────┐ │
│ │   Web Browser  │ │
│ └─▲──────────────┘ │
│   │ Open URL       │
│ ┌─┴──────────────┐ │
│ │  opener daemon │ │                 ┌────────────────────┐
│ └─┬──────────────┘ │                 │                    │
│   │                │                 │ ┌────────────────┐ │
│ ┌─┴──────────────┐ │   TCP request   │ │  open command  │ │
│ │  0.0.0.0:8080  │ ◄─────────────────┼─┤     (fake)     │ │
│ └────────────────┘ │                 │ └────────────────┘ │
│                    │                 │                    │
└────────────────────┘                 └────────────────────┘
       localhost                             container
```

Indeed, it seems that TCP support is needed in this case. If the container is used on host network, opener can listen on `127.0.0.1`. Listening on `0.0.0.0` seems risky (although there will be FW).

**Comment by mvgijssel** [comment]
> I can provide support for TCP, but first I want to clarify what you are trying to achieve. Do you want to use open command in a container to open a URL in the host machine's browser?

Yes! 

> Indeed, it seems that TCP support is needed in this case. If the container is used on host network, opener can listen on 127.0.0.1. Listening on 0.0.0.0 seems risky (although there will be FW).

Not very familiar with Go, but when opening a TCP socket it's often possible to provide a bind address. So you could also provide (an optional) bind parameter in the CLI which defaults to `127.0.0.1`, in case someone needs `0.0.0.0` or another address. 

BTW I have to say I'm impressed by your ASCI diagram 😱.

**Comment by superbrothers** [comment]
You too can easily create ASCII diagrams from today 😊

- https://asciiflow.com

**Comment by superbrothers** [comment]
I've released opener v0.1.0 included TCP support. Please try it!
